src/main/java/com/hackoeur/jglm/Mat.java

```diff
@@ -41,14 +41,14 @@
  * 
  * @author James Royalty
  */
-public interface Mat {
+public interface Mat<T extends Vec> {
```

Comment:
Because else a hidden but explicit cast happens when you get a column from a matrix. Not only do you need to manually cast it when the compiler can not work out to which vector type you're assigning, it's also extra runtime overhead (because the jvm has to check if the cast is valid).

Admitted, you don't *need* to parameterize the interface, but if you want to fix above mentioned situation, you'll have to replace all 'T' with the correct vector type directly in the matrix itself.